vertical-pod-autoscaler/recommender/model/cluster_test.go

```diff
@@ -104,45 +108,32 @@ func addTestPod(cluster *ClusterState) *PodState {
 	return cluster.Pods[testPodID]
 }
 
-// Creates a VPA followed by a matching pod. Verifies that the links between the
-// VPA and the pod are set correctly.
-func TestAddVpaThenAddPod(t *testing.T) {
-	cluster := NewClusterState()
-	vpa := addTestVpa(cluster)
-	pod := addTestPod(cluster)
-	assert.Equal(t, pod, vpa.Pods[testPodID])
-	assert.Equal(t, map[VpaID]*Vpa{testVpaID: vpa}, pod.MatchingVpas)
-}
-
-// Creates a pod followed by a matching VPA. Verifies that the links between the
-// VPA and the pod are set correctly.
-func TestAddPodThenAddVpa(t *testing.T) {
-	cluster := NewClusterState()
-	pod := addTestPod(cluster)
-	vpa := addTestVpa(cluster)
-	assert.Equal(t, pod, vpa.Pods[testPodID])
-	assert.Equal(t, map[VpaID]*Vpa{testVpaID: vpa}, pod.MatchingVpas)
+func addTestContainer(cluster *ClusterState) *ContainerState {
+	cluster.AddOrUpdateContainer(testContainerID, testRequest)
+	return cluster.GetContainer(testContainerID)
 }
 
-// Creates a VPA and a matching pod. Verifies that after deleting the VPA the
-// pod does not link to any Vpa.
-func TestDeleteVpa(t *testing.T) {
+// Creates a VPA followed by a matching pod. Verifies that the links between
+// VPA, the container and the aggregation are set correctly.
+func TestAddVpaThenAddPod(t *testing.T) {
 	cluster := NewClusterState()
 	vpa := addTestVpa(cluster)
-	pod := addTestPod(cluster)
-	cluster.DeleteVpa(vpa.ID)
-	assert.Empty(t, pod.MatchingVpas)
-	assert.NotContains(t, cluster.Vpas, vpa.ID)
+	addTestPod(cluster)
+	container := addTestContainer(cluster)
+	aggregateStateKey := cluster.aggregateStateKeyForContainerID(testContainerID)
+	assert.True(t, container.aggregator == vpa.aggregateContainerStates[aggregateStateKey])
```

Comment:
assert.Equal / assert.NotEqual won't work? xN

Reply:
This is slightly tricky, assert.Equal checks that values are equal, this asserts that the references are identical (https://godoc.org/github.com/stretchr/testify/assert#Assertions.Equal).